crates/memory/src/memory_manager/il2cpp.rs

```diff
@@ -0,0 +1,48 @@
+use crate::game_engine::il2cpp::*;
+use crate::process::Process;
+
+#[derive(Default)]
+pub struct UnityMemoryManager {
+    pub class: Option<Class>,
+    pub parent: Option<Class>,
+    pub instance: Option<u32>,
+    pub singleton: Option<Class>,
+    pub static_table: Option<u64>,
+}
+impl UnityMemoryManager {
+    pub fn reset(&mut self) {
+        *self = Self::default();
+    }
+}
+
+impl UnityMemoryManagement for UnityMemoryManager {
+    fn update(&mut self, process: &Process, module: &Module, image: &Image, name: &str) {
```

Comment:
So this one both initializes and acts as recover-from-reset?

Aside from that, does the code do anything frame-to-frame or is it only triggered after a reset?

Reply:
basically this files job is to make sure all those fields are available and if they are get the singleton. '

We probably dont need to query the singleton on every frame, so its something to optimize for sure - right now its the safest way before doing update_memory we have to avoid relying on memory errors.

Reply:
Will everything be available after a single call, or does it require multiple update cycles to get the entire context up and running?

Reply:
single call, but when changing screens pointers can drop, and in some cases we may get a new instance later

For example, player party manager's singleton instance is booted up at a different address every time you load the game. The entry to it is the same (parent static field "instance") but the instance is different.